compressor/compressor.cpp

```diff
@@ -0,0 +1,177 @@
+#include <string>
+#include <fstream>
+#include <deque>
+#include <iostream>
+#include "Bit_cashier.cpp"
+//Solve the issue with weird representation of first bytes in a text file
+//
+const int window_width = 510;//Won't work if more
+const int byte_length = 8;
+const int encoded_length = 17;
+const int untouched_length = 9;
+using Byte = char;
+
+
+void put_error_and_exit(const std::string& message)
+{
+  std::cerr << message;
+  exit(1);
+}
+
+int slide_window(int index, std::deque<Byte>& window, std::ifstream& in, std::streampos& end_of_file)
+{
+  while (index > window.size() / 2 && window.size() > 255)
+ {
+    if (in.tellg() < end_of_file)
+    {
+      window.push_back(in.get());
+      window.pop_front();
+      --index;
+    }
+    else
+    {
+      window.pop_front();
+      --index;
+    }
+  }
+
+  return index;
+}
+
+bool find_a_match(std::deque<Byte>& window, int i, int* offset_and_length)
```

Comment:
change int* offset_and_length to OffsetAndLength&
It going to be more readable and safe.